app/src/main/java/org/mozilla/fenix/utils/OnVisualCompletenessExecutionQueue.kt

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import kotlinx.coroutines.Job
+import mozilla.components.support.base.log.Log
+
+/**
+ * A queue of paused `Job`s whose execution is resumed when Fenix is visually complete.
+ *
+ * This queue contains paused `Job`s. Each Job in the queue will be restarted once Fenix
+ * is visually complete (as long as it has not be cancelled since it was added to the queue).
+ * Each Job will execute on the thread on which it was originally scheduled.
+ *
+ * This is threadsafe.
+ */
+object OnVisualCompletenessExecutionQueue {
```

Comment:
I spoke to you on Slack but I think this is more complex than we need: I think we really only need to hard-code methods to call like:
```kotlin
fun onVisualCompleteness() {
    GleanMetricsService.start()
    LeanplumMetricsService.start()
    // ...
}
```

I'm concerned about adding something more complicated than we need because it's harder to maintain, harder to make changes to, and more likely to be hiding bugs.